osbrain/agent.py

```diff
@@ -933,29 +944,82 @@ def _handle_async_requests(self, data):
         else:
             handler(self, response)
 
-    def _subscribe(self, alias: str, handlers: Dict[Union[bytes, str], Any]):
+    def subscribe(self, alias: str,
+                  handler: Dict[Union[bytes, str], Any]) -> None:
         """
-        Subscribe the agent to another agent.
+        Subscribe a SUB/SYNC_SUB socket given by its alias to the given
+        topics, and leave the handlers prepared internally.
 
         Parameters
         ----------
         alias
             Alias of the new subscriber socket.
-        handlers
+        handler
             A dictionary in which the keys represent the different topics
             and the values the actual handlers. If, instead of a dictionary,
             a single handler is given, it will be used to subscribe the agent
             to any topic.
         """
-        if not isinstance(handlers, dict):
-            handlers = {'': handlers}
+        if not isinstance(handler, dict):
+            handler = {'': handler}
+
+        curated_handlers = topics_to_bytes(handler)
 
-        curated_handlers = topics_to_bytes(handlers)
         # Subscribe to topics
-        for topic in curated_handlers.keys():
-            self.socket[alias].setsockopt(zmq.SUBSCRIBE, topic)
+        for topic in curated_handlers:
+            self._subscribe_to_topic(alias, topic)
+
         # Reset handlers
-        self._set_handler(self.socket[alias], curated_handlers)
+        if isinstance(self.address[alias], AgentChannel):
+            channel = self.address[alias]
+            sub_address = channel.receiver
+            uuid = channel.uuid
+            curated_handlers = topics_to_bytes(handler, uuid=uuid)
+            self._set_handler(self.socket[sub_address], curated_handlers,
+                              update=True)
+        else:
+            self._set_handler(self.socket[alias], curated_handlers,
+                              update=True)
+
+    def unsubscribe(self, alias: str, topic: Union[bytes, str]) -> None:
+        '''
+        Unsubscribe a SUB/SYNC_SUB socket given by its alias from a given
+        specific topic, and delete its entry from the handlers dictionary.
+        '''
+        if isinstance(topic, (tuple, list, dict)):
+            for t in topic:
+                self.unsubscribe(alias, t)
+            return
```

Comment:
This is not currently tested. Could you add a test? And also maybe update the docstring to say that a list/tuple is accepted in order to unsubscribe from multiple topics.